### PR TITLE
build(styled-components): Bump to version 4

### PR DESCRIPTION
**Pull request by danielbeardsley**
In most of the places where we depend on this, we're using
styled-components v4, so let's do the same here.

### CODE REVIEW

**Comment by sterlinghirsh** [comment]
CR :zap:

**Comment by djmetzle** [comment]
CR :shipit:

**Comment by addison-grant** [comment]
For QA, seems like at minimum it might be reasonable to clone the repo and run in dev mode https://github.com/iFixit/toolbox#development-setup and go through the components.

**Comment by addison-grant** [comment]
Can we test in iFixit with this change?

https://github.com/iFixit/ifixit/blob/master/UI/templates/js/package.json#L11
```diff
-      "@ifixit/toolbox": "^0.12.2",
+     "@ifixit/toolbox": "https://github.com/iFixit/toolbox.git#611d87ba32c6e166fb40411655363209b9d4a001"
" 
```

https://docs.npmjs.com/files/package.json#git-urls-as-dependencies

**Comment by sterlinghirsh** [comment]
There's an easier way using npm link

**Comment by danielbeardsley** [comment]
I'd opt for not testing it. This isn't a functional change.  i.e. The resulting npm package isn't different at all (except which version of the "peer dependency" is *listed*).

That said, we'll have a much bigger pull soon that *should* be tested that changed our build process.

**Comment by addison-grant** [comment]
I _think_ I see what you mean. I didn't know about peer dependencies. Do they just cause a failure if a needed package isn't installed by another dependency?

Aside: seems like you could get away with `^4` instead of `^4.0.0`.

File: package.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
     "react-dom": "^16.4.1",
     "react-test-renderer": "^16.4.1",
     "semantic-release": "^12.4.1",
-    "styled-components": "^3.3.3",
+    "styled-components": "^4.0.0",
```
Comment (addison-grant) [inline package.json]: This is not a peer dependency?

Comment (sterlinghirsh) [reply package.json]: Good catch, I think we will fix in a little bit in a separate pull (this one is merged now so oops)

**Pull request merged**